### DISCUSSION TITLE
funcion fail after insert "null value in column "id" "

### DISCUSSION THREAD

**Discussion by lawplatform** [supabase/supabase#18636] [Questions] [answered]
Currently, I'm using next_auth, and 
I want a function to be called in the public.users table 
whenever a new value is added to the next_auth.user table. 

Specifically, I'd like to copy the 'id' and 'email' from the next_auth.user table to the public.users table.

To achieve this, I've created the 'users' table with the 'id' column as both a primary and foreign key. I've also created the following function:
```
CREATE OR REPLACE FUNCTION create_profile_from_next_auth() 
RETURNS trigger AS $$ 
BEGIN 
	INSERT INTO public.users (id, email) 
	VALUES (NEW.id, NEW.email); 
	
	RETURN NEW; 
	END; 
$$ LANGUAGE plpgsql;
```
Additionally, 
I've created a trigger set to execute 'AFTER INSERT' on the 'next_auth.users' table.

However, when I attempt to add a new value to the 'next_auth.user' table, 
I encounter the following error:

```
ERROR:  23502: null value in column "id" of relation "users" violates not-null constraint
DETAIL:  Failing row contains (null, null, null, null, null).
CONTEXT:  SQL statement "INSERT INTO public.users (id, email) 
	VALUES (NEW.id, NEW.email)"
PL/pgSQL function create_profile_from_next_auth() line 3 at SQL statement
```

I'm not sure what's causing this issue, 
and I've tried searching for a solution without success.
Can you please help me understand what might be wrong?

**Comment by GaryAustin1** [comment, marked as answer]
Well new.id is null.  It could be you used a statement type trigger instead of a row trigger.
![image](https://github.com/supabase/supabase/assets/54564956/d327799b-cf9d-4fda-ae67-a61d5867126d)

**Reply by lawplatform** [reply]
thanks :) it solved my problem :)
